subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -323,6 +323,52 @@ static inline bool is_handshake_complete(struct tls_context *ctx)
 	return k_sem_count_get(&ctx->tls_established) != 0;
 }
 
+#if defined(CONFIG_NET_SOCKETS_TLS_SET_MAX_FRAGMENT_LENGTH) &&	\
+	defined(MBEDTLS_SSL_MAX_FRAGMENT_LENGTH) &&		\
+	(MBEDTLS_TLS_EXT_ADV_CONTENT_LEN < 16384)
```

Comment:
Wouldn't `MBEDTLS_TLS_EXT_ADV_CONTENT_LEN` resolve to 0 at this point given it's defined just below? I did some trials and it seems to be the case.

Reply:
Good catch, thanks. I've moved it above `#if` statement.